bin/barclamp_lib.rb

```diff
@@ -526,17 +237,206 @@ def check_user_password
   end
 end
 
+def deprecated_message(msg)
+  if $stdout.tty?
+    $stderr.puts "\e[31m#{msg}\e[0m"
+  else
+    $stderr.puts msg
+  end
+end
+
+def deprecated_exec(*cmd)
+  execute = [
+    "crowbarctl"
+  ].concat(
+    cmd
+  )
+
+  if @credentials
+    if @username.present?
+      execute.push "-U '#{@username}'"
+    end
+
+    if @password.present?
+      execute.push "-P '#{@password}'"
+    end
+  else
+    unless ENV["CROWBAR_KEY"].present?
+      execute.push "--anonymous"
+    end
+  end
+
+  if @timeout != 500
+    execute.push "-t #{@timeout}"
+  end
+
+  if @hostname != "127.0.0.1" || @port != 80
+    execute.push "-s 'http://#{@hostname}:#{@port}'"
+  end
+
+  if @username.empty? && @password.empty?
+    execute.push "--anonymous"
+  end
+
+  if @debug
+    execute.push "--debug"
+  end
+
+  deprecated_message <<-MSG.strip_heredoc
+    This command is deprecated, please use:
+    #{execute.join(" ")}
+  MSG
+
+  exec(execute.join(" "))
+end
+
 def run_sub_command(cmds, subcmd)
+  if [
+    "api_help",
+    "transition",
+    "show",
+    "delete",
+    "list",
+    "elements",
+    "element_node",
+    "proposal"
+  ].include?(subcmd)
+    case subcmd
+    when "api_help"
+      res = deprecated_exec(
+        "server",
+        "api",
+        @barclamp
+      )
+    when "transition"
+      usage(-2) unless ARGV.length == 2
+
+      res = deprecated_exec(
+        "node",
+        "transition",
+        ARGV.shift,
+        ARGV.shift
+      )
+    when "show"
+      usage(-2) unless ARGV.length == 1
+
+      res = deprecated_exec(
+        "proposal",
+        "show",
+        @barclamp,
+        ARGV.shift,
+        "--format json"
+      )
+    when "delete"
+      usage(-2) unless ARGV.length == 1
+
+      res = deprecated_exec(
+        "proposal",
+        "delete",
+        @barclamp,
+        ARGV.shift
+      )
+    when "list"
+      usage(-2) unless ARGV.length == 0
+
+      res = deprecated_exec(
+        "proposal",
+        "list",
+        @barclamp,
+        "--format plain"
+      )
+    when "elements"
+      usage(-2) unless ARGV.length == 0
+
+      res = deprecated_exec(
+        "role",
+        "list",
+        @barclamp,
+        "--format plain"
+      )
+    when "element_node"
+      usage(-2) unless ARGV.length == 1
+
+      res = deprecated_exec(
+        "role",
+        "show",
+        @barclamp,
+        ARGV.shift,
+        "--format plain"
+      )
+    when "proposal"
+      sub = ARGV.shift
+
+      case sub
+      when "commit"
+        args_count = 1
+        args_append = ""
+      when "dequeue"
+        args_count = 1
+        args_append = ""
+      when "delete"
+        args_count = 1
+        args_append = ""
+      when "create"
+        args_count = 1
+        args_append = ""
+
+        args_append.concat(
+          "--data '#{@data}'"
+        ) if @data.present?
+
+        args_append.concat(
```

Comment:
Metrics/BlockNesting: Avoid more than 3 levels of block nesting. (https://github.com/bbatsov/ruby-style-guide#three-is-the-number-thou-shalt-count)